org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -203,18 +210,20 @@ class Translator {
                         // We already have this data, so no need to make an unnecesary request over the wire.
                         return OpenT2T.invokeMethodAsync(translator, opent2tInfo.schema, 'get', [expand, winkDevice])
                             .then((platformResponse) => {
-                                return platformResponse; 
+                                return platformResponse;
                             });
                     }));
             }
         });
 
         // Return a promise for all platform translations.
-        return Promise.all(platformPromises)
-            .then((platforms) => {
-                var toReturn = {};
-                toReturn.schema = "org.opent2t.sample.hub.superpopular";
-                toReturn.platforms = platforms;
+        return Promise.all(platformPromises.map(promiseReflect))
```

Comment:
this is where the promiseReflect module comes into play (.map(promiseReflect)).  It catches failures, and resolves the promise as a success (so promise.all doesn't kick out), but still records the error for me to strip out below.

---
In reply to: [107274687](https://github.com/openT2T/translators/pull/216#discussion_r107274687) [](ancestors = 107274687)